### ISSUE TITLE
NuGet package leads to null reference in @Html.Kentico().PageBuilderScripts()

### ISSUE THREAD

**Issue by rkeitel** [Kentico/ems-mvc-components#83]
### Description

After installing the RichText NuGet package, @Html.Kentico().PageBuilderScripts() raises "Object reference not set to an instance of an object". After uninstalling the package and removing all of its files from wwwroot, the page builder correctly works again.

### Thoughts

The optimization (System.Web.Optimization) of the inline editor scripts doesn't work correctly.

### Links

[Kentico DevNet](https://devnet.kentico.com/questions/null-reference-from-@html-kentico()-pagebuilderscripts())

### Stacktrace

```
[NullReferenceException: Object reference not set to an instance of an object.]
   Microsoft.Ajax.Utilities.Member.GetFunctionGuess(AstNode target) +33
   Microsoft.Ajax.Utilities.AnalyzeNodeVisitor.GuessAtName(AstNode node) +117
   Microsoft.Ajax.Utilities.AnalyzeNodeVisitor.Visit(FunctionObject node) +388
   Microsoft.Ajax.Utilities.FunctionObject.Accept(IVisitor visitor) +35
   Microsoft.Ajax.Utilities.AnalyzeNodeVisitor.Visit(BinaryOperator node) +73
   Microsoft.Ajax.Utilities.AnalyzeNodeVisitor.Visit(Block node) +860
   Microsoft.Ajax.Utilities.AnalyzeNodeVisitor.Visit(FunctionObject node) +1467
   Microsoft.Ajax.Utilities.FunctionObject.Accept(IVisitor visitor) +35
   Microsoft.Ajax.Utilities.TreeVisitor.Visit(AstNodeList node) +169
   Microsoft.Ajax.Utilities.TreeVisitor.Visit(ArrayLiteral node) +43
   Microsoft.Ajax.Utilities.ArrayLiteral.Accept(IVisitor visitor) +35
   Microsoft.Ajax.Utilities.TreeVisitor.Visit(AstNodeList node) +169
   Microsoft.Ajax.Utilities.TreeVisitor.Visit(CallNode node) +56
   Microsoft.Ajax.Utilities.AnalyzeNodeVisitor.Visit(CallNode node) +2661
   Microsoft.Ajax.Utilities.CallNode.Accept(IVisitor visitor) +35
   Microsoft.Ajax.Utilities.AnalyzeNodeVisitor.Visit(Block node) +860
   Microsoft.Ajax.Utilities.JSParser.InternalParse() +2529
   Microsoft.Ajax.Utilities.Minifier.MinifyJavaScript(String source, CodeSettings codeSettings) +805
   System.Web.Optimization.JsMinify.Process(BundleContext context, BundleResponse response) +115
   System.Web.Optimization.Bundle.ApplyTransforms(BundleContext context, String bundleContent, IEnumerable`1 bundleFiles) +325
   System.Web.Optimization.Bundle.GetBundleResponse(BundleContext context) +59
   System.Web.Optimization.BundleResolver.GetBundleContents(String virtualPath) +213
   System.Web.Optimization.AssetManager.DeterminePathsToRender(IEnumerable`1 assets) +408
   System.Web.Optimization.AssetManager.RenderExplicit(String tagFormat, String[] paths) +34
   Kentico.PageBuilder.Web.Mvc.PageBuilderAssetsProvider.GetPageComponentsAdminScriptBundleTag() +157
   Kentico.PageBuilder.Web.Mvc.HtmlHelperExtensions.PageBuilderScriptsInternal(ExtensionPoint`1 instance, IAllowedDomainsRetriever retriever, IComponentLocalizationService localizationService) +798
   ASP._Page_Views_Shared_WebPage_cshtml.<Execute>b__3_1() in C:\inetpub\wwwroot\staff-k12\Staff\Views\Shared\WebPage.cshtml:8
   System.Web.WebPages.<>c__DisplayClass7.<RenderSection>b__5(TextWriter tw) +414
   System.Web.WebPages.WebPageBase.Write(HelperResult result) +108
   ASP._Page_Views_Shared_Layouts_Page_cshtml.<Execute>b__3_1() in C:\inetpub\wwwroot\staff-k12\Staff\Views\Shared\Layouts\Page.cshtml:12
   System.Web.WebPages.<>c__DisplayClass7.<RenderSection>b__5(TextWriter tw) +414
   System.Web.WebPages.WebPageBase.Write(HelperResult result) +108
   ASP._Page_Views_Shared_Layouts_PageOuter_cshtml.Execute() in C:\inetpub\wwwroot\staff-k12\Staff\Views\Shared\Layouts\PageOuter.cshtml:27
   System.Web.WebPages.WebPageBase.ExecutePageHierarchy() +252
   System.Web.Mvc.WebViewPage.ExecutePageHierarchy() +148
   System.Web.WebPages.WebPageBase.ExecutePageHierarchy(WebPageContext pageContext, TextWriter writer, WebPageRenderingBase startPage) +122
   System.Web.WebPages.<>c__DisplayClass3.<RenderPageCore>b__2(TextWriter writer) +309
   System.Web.WebPages.WebPageBase.Write(HelperResult result) +108
   System.Web.WebPages.WebPageBase.RenderSurrounding(String partialViewName, Action`1 body) +89
   System.Web.WebPages.WebPageBase.PopContext() +310
   System.Web.WebPages.<>c__DisplayClass3.<RenderPageCore>b__2(TextWriter writer) +309
   System.Web.WebPages.WebPageBase.Write(HelperResult result) +108
   System.Web.WebPages.WebPageBase.RenderSurrounding(String partialViewName, Action`1 body) +89
   System.Web.WebPages.WebPageBase.PopContext() +310
   System.Web.Mvc.ViewResultBase.ExecuteResult(ControllerContext context) +378
   System.Web.Mvc.ControllerActionInvoker.InvokeActionResultFilterRecursive(IList`1 filters, Int32 filterIndex, ResultExecutingContext preContext, ControllerContext controllerContext, ActionResult actionResult) +91
   System.Web.Mvc.ControllerActionInvoker.InvokeActionResultFilterRecursive(IList`1 filters, Int32 filterIndex, ResultExecutingContext preContext, ControllerContext controllerContext, ActionResult actionResult) +793
   System.Web.Mvc.ControllerActionInvoker.InvokeActionResultFilterRecursive(IList`1 filters, Int32 filterIndex, ResultExecutingContext preContext, ControllerContext controllerContext, ActionResult actionResult) +793
   System.Web.Mvc.ControllerActionInvoker.InvokeActionResultFilterRecursive(IList`1 filters, Int32 filterIndex, ResultExecutingContext preContext, ControllerContext controllerContext, ActionResult actionResult) +793
   System.Web.Mvc.ControllerActionInvoker.InvokeActionResultWithFilters(ControllerContext controllerContext, IList`1 filters, ActionResult actionResult) +81
   System.Web.Mvc.Async.<>c__DisplayClass21.<BeginInvokeAction>b__1e(IAsyncResult asyncResult) +188
   System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeAction(IAsyncResult asyncResult) +38
   System.Web.Mvc.Controller.<BeginExecuteCore>b__1d(IAsyncResult asyncResult, ExecuteCoreState innerState) +32
   System.Web.Mvc.Async.WrappedAsyncVoid`1.CallEndDelegate(IAsyncResult asyncResult) +73
   System.Web.Mvc.Controller.EndExecuteCore(IAsyncResult asyncResult) +52
   System.Web.Mvc.Async.WrappedAsyncVoid`1.CallEndDelegate(IAsyncResult asyncResult) +39
   System.Web.Mvc.Controller.EndExecute(IAsyncResult asyncResult) +38
   System.Web.Mvc.MvcHandler.<BeginProcessRequest>b__5(IAsyncResult asyncResult, ProcessRequestState innerState) +46
   System.Web.Mvc.Async.WrappedAsyncVoid`1.CallEndDelegate(IAsyncResult asyncResult) +73
   System.Web.Mvc.MvcHandler.EndProcessRequest(IAsyncResult asyncResult) +38
   System.Web.CallHandlerExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute() +651
   System.Web.HttpApplication.ExecuteStepImpl(IExecutionStep step) +220
   System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously) +134
```

**Comment by rpribyl** [comment]
Hi @rkeitel,
Thank you for the issue report. It is caused indeed by the .NET bundler's minification feature.

Disabling the minification and bundling .NET feature is a possible workaround for now.
`BundleTable.EnableOptimizations = false`

We'll try to address the issue as soon as possible.

**Comment by rkeitel** [comment]
Hi,
Thank you for the quick response. Sadly your workaround doesn't work :-(.

**Comment by rpribyl** [comment]
That is strange, this line of code should disable the bundler together with the minifier.
Did you place it into the _Global.asax.cs_ into the _Application_Start_ method?

**Comment by rkeitel** [comment]
Yes, I did.

**Comment by rkeitel** [comment]
Do you have another idea?

**Comment by giorgioLUK** [comment]
I am afraid we would need backups of your project and database to further investigate it. 

To use our FTP, please follow these instructions:

Compress the backup files to a ZIP archive using WinZip or 7-zip. If the ZIP file(s) are larger than 150 MB, you can split them into smaller files so that you can simultaneously upload more than one file. If there are many physical files that are not related to this issue (typically media files), feel free to remove them from the backup.
Connect using a client like FileZilla to the nearest server and the credentials below:

Western Europe: EU-west.ftp.kentico.com
U.S. East Coast: US-east.ftp.kentico.com
Username: kentico
Password: kenti38Co

Create a new folder on the FTP server using e.g. your nickname for the folder name. Make sure to enter the name when you create the folder, since for security reasons you are not allowed to modify folder names. If you make a mistake, just make a new folder and let us know.
Upload the ZIP file(s) into that folder and let us know when you are finished and what folder you used.

Thank you!

**Comment by rkeitel** [comment]
Hi,

sadly I cannot successfully connect to your FTP. I uploaded the files to my business OneDrive, there are no sensitive data:
- [Source code ZIP](https://mspartnerwebit-my.sharepoint.com/:u:/g/personal/keitel_webit_de/EUfVPlW1cWdBjdPpsjQtM5EBioQaCINbJnJ8BAdIjz2k2w?e=neTW3F)
- [Database backup file](https://mspartnerwebit-my.sharepoint.com/:u:/g/personal/keitel_webit_de/EWIN0b29p2hBkcrP9C53qKgBmOgRCZPmF-s4CV4Gnt4YVg?e=U1z9ZG)

I configured a new Kentico test project and install this NuGet package, there all is working fine. But I couldn't figure out, why it is not working in my project :-(

**Comment by petrinecp** [comment]
Hi @rkeitel, thank you for the backups I was able to reproduce the issue on your solution. I'll get back to you as soon as I'll have more info.

**Comment by petrinecp** [comment]
Hi @rkeitel a fix for this issue was released in 12.0.54 Hotfix, however the automatic js minification had to be turned off in order to make this work therefore if you want to have your js files minified in production, please include a `*.min.js` versions of you scripts.

**Comment by rkeitel** [comment]
Thanks, now it is working :-)
